app/controllers/learn_time_controller.rb

```diff
@@ -2,6 +2,7 @@ class LearnTimeController < ApplicationController
   def create
     @link = Link.find(params[:link_id])
     @learn_time = LearnTime.create!(user_id: current_user.id, link_id: params[:link_id])
+    @link.update_column(:last_learned_at, @learn_time.created_at)
```

Comment:
`LearnTime` creation and `last_learned_at` should be wrapped in a transaction.

Check http://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html